### PR TITLE
Create custom connector TrueDialog

### PR DESCRIPTION
**Pull request by TrueDialogDynamics**
- [x] I attest that the connector works and I verified by deploying and testing all the operations. 
- [x] I attest that I have added detailed descriptions for all operations and parameters in the swagger file.
- [x] I attest that I have added response schemas to my actions, unless the response schema is dynamic. 
- [x] I validated the swagger file, `apiDefinition.swagger.json`, by running `paconn validate` command.
- [x] If this is a certified connector, I confirm that `apiProperties.json` has a valid brand color and doesn't use an invalid brand color, `#007ee5` or `#ffffff`. If this is an independent publisher connector, I confirm that I am not submitting a connector icon.

If you are an Independent Publisher, you must also attest to the following to ensure a smooth publishing process:
- [ ] I have named this PR after the pattern of "Connector Name (Independent Publisher)" ex: HubSpot Marketing (Independent Publisher)
- [ ] Within this PR markdown file, I have pasted screenshots that show: 3 unique operations (actions/triggers) working within a Flow. This can be in one flow or part of multiple flows. For each one of those flows, I have pasted in screenshots of the Flow succeeding. 
- [ ] Within this PR markdown file, I have pasted in a screenshot from the Test operations section within the Custom Connector UI.
- [ ] If the connector uses OAuth, I have provided detailed steps on how to create an app in the readme.md.

### CODE REVIEW

File: certified-connectors/TrueDialog SMS/apiDefinition.swagger.json [hunk 1]
```diff
@@ -0,0 +1,1762 @@
...
+                  "type": "string",
+                  "description": "URL",
+                  "title": "URL",
+                  "x-ms-notification-url": true,
+                  "x-ms-visibility": "internal"
+                },
+                "Active": {
+                  "type": "boolean",
+                  "description": "Active",
+                  "title": "",
```
Comment (sriyen-msft) [inline certified-connectors/TrueDialog SMS/apiDefinition.swagger.json:155]: @TrueDialogDynamics Empty property values can be removed until unless you have a specific reason to keep them.

File: certified-connectors/TrueDialog SMS/apiDefinition.swagger.json [hunk 2]
```diff
@@ -0,0 +1,1762 @@
...
+              "description": "Unique token",
+              "title": "TransactionId"
+            }
+          }
+        }
+      },
+      "post": {
+        "responses": {
+          "201": {
+            "description": "Incoming messages callback created"
```
Comment (sriyen-msft) [inline certified-connectors/TrueDialog SMS/apiDefinition.swagger.json:98]: @TrueDialogDynamics Can you please let us know if this post action has schema definition?

File: certified-connectors/TrueDialog SMS/apiDefinition.swagger.json [hunk 3]
```diff
@@ -0,0 +1,1762 @@
...
+              "description": "SubscriptionId",
+              "x-ms-visibility": "internal"
+            }
+          }
+        }
+      },
+      "put": {
+        "responses": {
+          "201": {
+            "description": "Keyword callback created"
```
Comment (sriyen-msft) [inline certified-connectors/TrueDialog SMS/apiDefinition.swagger.json:269]: @TrueDialogDynamics Empty response schema is not allowed anymore, which means no dynamic content will be shown in the output and makers have to use parse JSON to parse the response. Please provide a valid response schema definition for all the operations in the swagger.

File: certified-connectors/TrueDialog SMS/apiDefinition.swagger.json [hunk 4]
```diff
@@ -0,0 +1,1762 @@
...
+              "description": "SubscriptionId",
+              "x-ms-visibility": "internal"
+            }
+          }
+        }
+      },
+      "put": {
+        "responses": {
+          "201": {
+            "description": "Stop callback created"
```
Comment (sriyen-msft) [inline certified-connectors/TrueDialog SMS/apiDefinition.swagger.json:429]: @TrueDialogDynamics Empty response schema is not allowed anymore, which means no dynamic content will be shown in the output and makers have to use parse JSON to parse the response. Please provide a valid response schema definition for all the operations in the swagger.

File: certified-connectors/TrueDialog SMS/apiDefinition.swagger.json [hunk 5]
```diff
@@ -0,0 +1,1762 @@
...
+          "type": "object",
+          "properties": {
+            "CallbackTimestamp": {
+              "type": "string",
+              "description": "Date and time the callback has been sent. Format: yyyy-MM-ddTHH:mm:ss",
+              "title": "Callback Timestamp"
+            },
+            "CallbackToken": {
+              "type": "string",
+              "description": "CallbackToken"
```
Comment (sriyen-msft) [inline certified-connectors/TrueDialog SMS/apiDefinition.swagger.json:518]: @TrueDialogDynamics All the description values of properties are expected to be free from grammatical and spell errors. Please provide valid description values.
Docs - [Define operation and parameter summaries and descriptions](https://docs.microsoft.com/connectors/custom-connectors/certification-submission#define-operation-and-parameter-summaries-and-descriptions)

File: certified-connectors/TrueDialog SMS/apiDefinition.swagger.json [hunk 6]
```diff
@@ -0,0 +1,1762 @@
...
+              "description": "The actual message text",
+              "title": "Message"
+            }
+          }
+        }
+      },
+      "put": {
+        "responses": {
+          "201": {
+            "description": "Delivery Notice callback created"
```
Comment (sriyen-msft) [inline certified-connectors/TrueDialog SMS/apiDefinition.swagger.json:597]: @TrueDialogDynamics Empty response schema is not allowed anymore, which means no dynamic content will be shown in the output and makers have to use parse JSON to parse the response. Please provide a valid response schema definition for all the operations in the swagger.

File: certified-connectors/TrueDialog SMS/apiDefinition.swagger.json [hunk 7]
```diff
@@ -0,0 +1,1762 @@
...
+              },
+              "description": "InvalidTargets"
+            }
+          }
+        }
+      },
+      "put": {
+        "responses": {
+          "201": {
+            "description": "Invalid Targets callback created"
```
Comment (sriyen-msft) [inline certified-connectors/TrueDialog SMS/apiDefinition.swagger.json:750]: @TrueDialogDynamics Empty response schema is not allowed anymore, which means no dynamic content will be shown in the output and makers have to use parse JSON to parse the response. Please provide a valid response schema definition for all the operations in the swagger.

File: certified-connectors/TrueDialog SMS/apiDefinition.swagger.json [hunk 8]
```diff
@@ -0,0 +1,1762 @@
...
+                  "type": "string",
+                  "description": "URL",
+                  "title": "URL",
+                  "x-ms-notification-url": true,
+                  "x-ms-visibility": "internal"
+                },
+                "Active": {
+                  "type": "boolean",
+                  "description": "Active",
+                  "title": "",
```
Comment (sriyen-msft) [inline certified-connectors/TrueDialog SMS/apiDefinition.swagger.json:807]: @TrueDialogDynamics Empty property values can be removed until unless you have a specific reason to keep them.

File: certified-connectors/TrueDialog SMS/apiProperties.json [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+            "tabIndex": 3,
+            "clearText": false,
+            "required": "true"
+          }
+        }
+      }
+    },
+    "iconBrandColor": "#ff7800",
+    "capabilities": [],
+    "publisher": "TrueDialog Dynamics"
```
Comment (sriyen-msft) [inline certified-connectors/TrueDialog SMS/apiProperties.json:33]: @TrueDialogDynamics Please add stackOwner property as well..

Documentation - [Add metadata](https://docs.microsoft.com/en-us/connectors/custom-connectors/certification-submission#step-3-add-metadata)

File: certified-connectors/TrueDialog SMS/readme.md [hunk 10]
```diff
@@ -0,0 +1,52 @@
+
+## The TrueDialog Connector 
+
+TrueDialog's SMS API is enterprise-grade with more pre-built features than any other SMS software. Built for developers and hosted in the Microsoft Azure cloud, our SMS API is highly reliable, secure and includes a pre-built ISO 9001 database for extra opt-out and compliance protection. Also available through our full-featured Enterprise-Grade SMS Texting Software.
+
+[Learn more](https://www.truedialog.com/sms-api-texting-api/) about what sets TrueDialog’s API apart from the rest.
+
```
Comment (sriyen-msft) [inline certified-connectors/TrueDialog SMS/readme.md:7]: @TrueDialogDynamics Please go thru the [template provided for readme.md](https://github.com/microsoft/PowerPlatformConnectors/blob/dev/templates/certified-connectors/readme.md) and add any required sections you are missing.

**Review by sriyen-msft — CHANGES_REQUESTED** [review]

File: certified-connectors/TrueDialog SMS/readme.md (hunk 10 above)
Comment (TrueDialogDynamics) [reply certified-connectors/TrueDialog SMS/readme.md:9]: Updated readme file according to the provided template in [this commit](https://github.com/microsoft/PowerPlatformConnectors/pull/1337/commits/3ab58071b349a4d70afcde2be80e5b110bef7be5)

File: certified-connectors/TrueDialog SMS/apiDefinition.swagger.json (hunk 6 above)
Comment (TrueDialogDynamics) [reply certified-connectors/TrueDialog SMS/apiDefinition.swagger.json]: Added schema for all the responses in [this commit](https://github.com/microsoft/PowerPlatformConnectors/pull/1337/commits/d4d6d4c7b8aa28c0eceb87f9eccd4625ec7c5d84)

File: certified-connectors/TrueDialog SMS/apiDefinition.swagger.json (hunk 4 above)
Comment (TrueDialogDynamics) [reply certified-connectors/TrueDialog SMS/apiDefinition.swagger.json]: Added schema for all the responses in [this commit](https://github.com/microsoft/PowerPlatformConnectors/pull/1337/commits/d4d6d4c7b8aa28c0eceb87f9eccd4625ec7c5d84)

File: certified-connectors/TrueDialog SMS/apiDefinition.swagger.json (hunk 3 above)
Comment (TrueDialogDynamics) [reply certified-connectors/TrueDialog SMS/apiDefinition.swagger.json]: Added schema for all the responses in [this commit](https://github.com/microsoft/PowerPlatformConnectors/pull/1337/commits/d4d6d4c7b8aa28c0eceb87f9eccd4625ec7c5d84)

File: certified-connectors/TrueDialog SMS/apiDefinition.swagger.json (hunk 2 above)
Comment (TrueDialogDynamics) [reply certified-connectors/TrueDialog SMS/apiDefinition.swagger.json]: @sriyen-msft  Added schema for all the responses in [this commit](https://github.com/microsoft/PowerPlatformConnectors/pull/1337/commits/d4d6d4c7b8aa28c0eceb87f9eccd4625ec7c5d84)

File: certified-connectors/TrueDialog SMS/apiDefinition.swagger.json (hunk 8 above)
Comment (TrueDialogDynamics) [reply certified-connectors/TrueDialog SMS/apiDefinition.swagger.json]: Fixed empty Titles in [this commit](https://github.com/microsoft/PowerPlatformConnectors/pull/1337/commits/25796ca1fe660681d7eddabb91e8614fbc19a1ef)

File: certified-connectors/TrueDialog SMS/apiDefinition.swagger.json (hunk 5 above)
Comment (TrueDialogDynamics) [reply certified-connectors/TrueDialog SMS/apiDefinition.swagger.json]: Fixed descriptions in [this commit](https://github.com/microsoft/PowerPlatformConnectors/pull/1337/commits/11f2fa878ad92d61167f1226749ff251ca69060b)

**Review by sriyen-msft — APPROVED** [review]

**Comment by sriyen-msft** [comment]
@TrueDialogDynamics Sorry for the delay in response, Your connector pull request is approved and merged, you can submit your connector artifacts to the Connector certification tab in ISV Studio to move forward in the certification process, Thanks.

**Pull request merged**